pkg/authenticator/authenticator.go

```diff
@@ -250,32 +285,31 @@ func (auth *Authenticator) Authenticate() ([]byte, error) {
 		return nil, log.RecordedError(log.CAKC027, err)
 	}
 
-	return DataResponse(resp)
+	err = validateResponse(resp)
+	if err != nil {
+		return nil, err
+	}
+
+	return readBody(resp)
 }
 
-// ParseAuthenticationResponse takes the response from the Authenticate
-// request, decrypts if needed, and writes to the token file
-func (auth *Authenticator) ParseAuthenticationResponse(response []byte) error {
+// parseAuthenticationResponse takes the response from the Authenticate
+// request, decrypts if needed, and returns it
+func (auth *Authenticator) parseAuthenticationResponse(response []byte) ([]byte, error) {
 	var content []byte
 	var err error
 
 	// Token is only encrypted in Conjur v4
 	if auth.Config.ConjurVersion == "4" {
 		content, err = decodeFromPEM(response, auth.PublicCert, auth.privateKey)
```

Comment:
why are we decoding only in v4 and not also in v5?

Reply:
i guess v4 returned the certificate encoded and v5 does not. i hope this code snippet can be removed soon

Reply:
we deprecated support for authn-k8s in [DAP 11.5](https://docs.cyberark.com/Product-Doc/OnlineHelp/PAS/Latest/en/Content/Release%20Notes/RN-WhatsNew11-5.htm?tocpath=Get%20Started%7CWhat%E2%80%99s%20New%7CRelease%20Notes%7C_____5)

I think we can remove support in the next release, since we already announced deprecation with 11.5. the changelog entry should probably reference the 11.5 release notes. you can see the [Puppet changelog](https://github.com/cyberark/conjur-puppet/blob/master/CHANGELOG.md) for example entries on removing support.

Reply:
Created an issue for this - https://github.com/cyberark/conjur-authn-k8s-client/issues/183